Change error message and add Open Google MC button to the Dashboard

tomalec opened the pull request:
### Changes proposed in this Pull Request:

- Change error message and add Open Google MC button, (8bc5c6a)
	In case of missing data dor Free Listings in the dashboard.
	Addresses 1. of https://github.com/woocommerce/google-listings-and-ads/issues/543#issuecomment-848436850
	Implements https://github.com/woocommerce/google-listings-and-ads/issues/708
- Align `PerformanceCard` and `PaidCampaignPromotionCard` visually, (44a571e)
	make them share CSS.
- Replace `SummaryCard`'s specific margins with gap property. (0a6c3bd)
	So the conatiner dictates its gutters, and we will not have to remember to attach `_text` classname, and assume `.components-button` is not the first children.
- Make `SummaryNumber` (& placeholder) adapt to a higher `SummaryCard` (b52bcc7)


### Screenshots:

<!--- Optional --->


### Detailed test instructions:

0. Reproduce the error scenario from https://github.com/woocommerce/google-listings-and-ads/issues/543 or mock the data with the following snippet in `google-listings-and-ads.php`
```php
add_filter(
	'gla_prepared_response_mc-reports-programs',
	function( $response ) {
		return [];
	},
);
```
1. Go to the Dashboard [`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard)
2. Wait for the request to fail
3. Check that message is shown correctly
4. Check that "Open Google Merchant Center"
	1. Takes you to Google's MC
	2. Fires tracking event once clicked


### Changelog Note:

> Added "Open Google Merchant Center" link to the dashboard error message

tomalec committed:
Change error message and add Open Google MC button,

In case of missing data dor Free Listings in the dashboard.

Addresses 1. of https://github.com/woocommerce/google-listings-and-ads/issues/543#issuecomment-848436850

tomalec committed:
Align `PerformanceCard` and `PaidCampaignPromotionCard` visually,

make them share CSS.

tomalec committed:
Replace `SummaryCard`'s specific margins with gap property.

So the contaienr dictate its gutters, and we will not have to remember to attach `_text` classname, and assume `.components-button` is not the first children.

tomalec committed:
Make `SummaryNumber` (&placeholder) adapt to a higher `SummaryCard`

tomalec committed:
Add `href` property to the `gla_google_mc_link_click` track.

eason9487 reviewed (approved):
Tested with the mocked data and simulated API error. LGTM.

tomalec committed:
Workaround lack of `gap` for flexbox in Safari by using grid layout.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/709#discussion_r643636489

eason9487 reviewed (approved):
Testing well and LGTM.
